app/code/Magento/ConfigurableProduct/Model/Product/Type/Configurable.php

```diff
@@ -585,7 +592,7 @@ protected function getGalleryReadHandler()
      * @param  \Magento\Catalog\Model\Product $product
      * @return \Magento\ConfigurableProduct\Model\ResourceModel\Product\Type\Configurable\Product\Collection
      */
-    public function getUsedProductCollection($product)
+    protected function getLinkedProductCollection($product)
```

Comment:
Hello and thank you for your contribution. Just a question: what's the reason behind the introduction of this new method?

Reply:
@aleron75 
as described in  #26065

the swatches plugin is not needed `Magento\Swatches\Model\Plugin\Configurable::afterGetUsedProductCollection`
so `isSaleable` can call internal method. It slightly improves performance if swatches are disabled on category page (listing page). Before, when the code called `isSaleable` it was always called `afterGetUsedProductCollection` that just adds attributes to select, however they are not needed in this check.
It also helps if `isSaleable` is called a few times in different places

Reply:
Sorry, @ilnytskyi I didn't explain myself. I got the point of the optimization, I was just wondering why did you introduce the additional `getLinkedProductCollection()` method.

Reply:
In general, to avoid code duplication inside `isSalable`. (collection building)
`getUsedProductCollection` is used inside `isSalable` so i decided to use it as wrapper for the collection building then move its logic to the new method. This way
`getUsedProductCollection` can be used as previously and
getLinkedProductCollection only internal method that is not affected by plugins

Reply:
ok great, thank you!